www/src/components/overview/clusters/plural-cloud/ConsoleInstanceOIDC.tsx

```diff
@@ -0,0 +1,217 @@
+  Accordion,
+  AccordionItem,
+  Button,
+  Flex,
+  FormField,
+  Modal,
+  PeopleIcon,
+} from '@pluralsh/design-system'
+import {
+  BindingInput,
+  fetchGroups,
+  fetchUsers,
+} from 'components/account/Typeaheads'
+import { SanitizePropsType, sanitize } from 'components/account/utils'
+import { UrlsInput } from 'components/app/oidc/OIDC'
+import { GqlError } from 'components/utils/Alert'
+import ImpersonateServiceAccount from 'components/utils/ImpersonateServiceAccount'
+import {
+  ConsoleInstanceFragment,
+  InputMaybe,
+  OidcAuthMethod,
+  OidcProviderBinding,
+  useRepositoryQuery,
+  useUpdateOidcProviderMutation,
+} from 'generated/graphql'
+import { isEmpty } from 'lodash'
+import { useState } from 'react'
+import { useTheme } from 'styled-components'
+
+export function ConsoleInstanceOIDC({
+  instance,
+}: {
+  instance: ConsoleInstanceFragment
+}) {
+  const [open, setOpen] = useState(false)
+
+  return (
+    <ImpersonateServiceAccount
+      id={instance.console?.owner?.id}
+      renderIndicators={false}
+    >
+      <>
+        <Button
+          secondary
+          startIcon={<PeopleIcon />}
+          onClick={() => setOpen(true)}
+        >
+          OIDC
+        </Button>
+        <Modal
+          onOpenAutoFocus={(e) => e.preventDefault()}
+          open={open}
+          onClose={() => setOpen(false)}
+          size="large"
+        >
+          <ConsoleInstanceOIDCInner
+            instance={instance}
+            onClose={() => setOpen(false)}
+          />
+        </Modal>
+      </>
+    </ImpersonateServiceAccount>
+  )
+}
+
+function ConsoleInstanceOIDCInner({
+  instance,
+  onClose,
+}: {
+  instance: ConsoleInstanceFragment
+  onClose: () => void
+}) {
+  const [bindings, setBindings] = useState<any>([])
+  const [redirectUris, setRedirectUris] = useState<InputMaybe<string>[]>([])
+
+  const {
+    data,
+    loading: loadingRepo,
+    error: errorRepo,
+  } = useRepositoryQuery({
+    variables: { name: 'console' },
+    fetchPolicy: 'cache-and-network',
+    onCompleted: (data) => {
+      setBindings(data?.repository?.installation?.oidcProvider?.bindings)
+      setRedirectUris(
+        data?.repository?.installation?.oidcProvider?.redirectUris ?? []
+      )
+    },
+  })
+
+  const installation = data?.repository?.installation
+  const provider = installation?.oidcProvider
+
+  const [mutation, { loading: loadingMutation, error: errorMutation }] =
+    useUpdateOidcProviderMutation({
+      variables: {
+        id: installation?.id ?? '',
+        attributes: {
+          authMethod: provider?.authMethod ?? OidcAuthMethod.Post,
+          redirectUris: isEmpty(redirectUris)
+            ? [`https://${instance.url}/oauth/callback`]
+            : redirectUris,
+          bindings: bindings.map((value) =>
+            sanitize(value as SanitizePropsType)
+          ),
+        },
+      },
+      onCompleted: onClose,
+    })
+
+  if (!provider?.bindings || loadingRepo || errorRepo) return null
```

Comment:
nit: There is this initial dialog flicker when initializing the data where it renders an empty dialog first. We can either render a normal-sized dialog and show a loading spinner or wrap the dialog in a suspense that can fallback to the loading overlay until the data is loaded.

![image](https://github.com/user-attachments/assets/eea6452f-4ec2-4ea3-9b04-0a2712aab73d)

Reply:
fixed, went with the suspense option i think it looks good